Kitodo/src/main/java/org/kitodo/services/image/ImageGenerator.java

```diff
@@ -0,0 +1,423 @@
+        boolean isCreatingDerivative = imageProperties.getDerivative().isPresent();
+        boolean isChangingDpi = imageProperties.getDpi().isPresent();
+        boolean isGettingScaledWebImage = imageProperties.getImageScale().isPresent();
+        boolean isGettingSizedWebImage = imageProperties.getImageSize().isPresent();
+
+        if (isCreatingDerivative) {
+            createDerivative(sourceImage, imageProperties, destinationImage.getFileFormat().getImageFileFormat().get(),
+                destinationImage.getUri(canonical));
+        } else if (isChangingDpi || isGettingScaledWebImage || isGettingSizedWebImage) {
+            createImageWithImageIO(sourceImage, imageProperties, destinationImage.getFileFormat(),
+                destinationImage.getUri(canonical));
+        }
+    }
+
+    /**
+     * Returns from toBeGenerated the item specified by position.
+     *
+     * @return the item specified by position
+     */
+    Pair<Pair<String, URI>, List<Subfolder>> getFromToBeGeneratedByPosition() {
+        return toBeGenerated.get(position);
+    }
+
+    /**
+     * Returns the current position in the list.
+     *
+     * @return the position
+     */
+    int getPosition() {
+        return position;
+    }
+
+    /**
+     * Returns the list of source images.
+     *
+     * @return the list of source images
+     */
+    List<Pair<String, URI>> getSources() {
+        return sources;
+    }
+
+    /**
+     * Returns the list of image generation task descriptions.
+     *
+     * @return the list of image generation task descriptions
+     */
+    List<Pair<Pair<String, URI>, List<Subfolder>>> getToBeGenerated() {
+        return toBeGenerated;
+    }
+
+    /**
+     * Returns the enum constant inicating the variant of the image generator
+     * task.
+     *
+     * @return the variant of the image generator task
+     */
+    GenerationMode getMode() {
+        return mode;
+    }
+
+    /**
+     * Lets the supervisor do something, if there is one. Otherwise nothing
+     * happens.
+     *
+     * @param action
+     *            what the supervisor should do
+     */
+    void letTheSupervisor(Consumer<EmptyTask> action) {
+        if (supervisor.isPresent()) {
+            action.accept(supervisor.get());
+        }
+
+    }
+
+    /**
+     * Invokes one of the three methods of the image management interface that
+     * return a Java image. Which method is called and its parameters are
+     * determined in the configuration of the folder.
+     *
+     * @param sourceImage
+     *            address of the source image from which the derivative is to be
+     *            calculated.
+     * @param imageProperties
+     *            configuration for the target image
+     * @return an image in memory
+     * @throws IOException
+     *             if an underlying disk operation fails
+     */
+    private static Image retrieveJavaImage(URI sourceImage, Folder imageProperties) throws IOException {
```

Comment:
Why is it explicitly "JavaImage" istead of Image?

Reply:
Because these methods return a java image object in memory. In contrast, the image created by createDerivative is not a Java image (object), but only an image (written to disk). I wanted to better distinguish the two things with this choice of words.